### PR TITLE
ES #498: http2 translated to Spanish

### PR DESCRIPTION
**Pull request by moniloria**
In some links I added hreflang="en" and for some words,  lang="en" inside the italics tag

Makes progress on #498

This is my first chapter, please if you have any comments, I would appreciate it.

I also found some errors in the source content in English, who should I tell?

Thanks!

### CODE REVIEW

**Comment by bazzadp** [comment]
Thanks - will check it out!

> I also found some errors in the source content in English, who should I tell?

If they are small and you're reasonably certain they are errors (e.g. typos) can you add to this pull request? If you've questions or want to clarify something is correct [can you raise an issue](https://github.com/HTTPArchive/almanac.httparchive.org/issues) to discuss first?

File: src/content/es/2020/http2.md [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+  sql_file="http2_1st_party_vs_3rd_party.sql",
+  sheets_gid="733872185"
+)
+}}
+¿Hay alguna diferencia ente cuáles tipos de contenido se sirven a través de HTTP/2 o gQUIC? La Figura 22.8 muestra, por ejemplo, que el 90% de los sitios web sirven el 100% de fuentes y audio de terceros a través de HTTP/2 o gQUIC, solo el 5% a través de HTTP/1.1 y un 5% son una combinación. La mayoría de los activos de terceros son scripts o imágenes, y se sirven únicamente a través de HTTP/2 o gQUIC en el 60% y el 70% de los sitios web, respectivamente.
+
+{{ figure_markup(
+  image="http2-third-party-http2-usage-by-content-type.png",
+  caption="Fracción de solicitudes HTTP/2 o gQUIC de terceros conocidos por <i lang="en">content-type</i> por sitio web.",
+  description="Un gráfico de barras que compara la fracción de solicitudes HTTP/2 de terceros por <i lang="en">content-type</i>. Todas las solicitudes de terceros se envían a través de HTTP/2 o gQUIC para el 90% de audio y fuentes, 80% de css y video, 70% de html, imágenes y texto y 60% de scripts.",
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:203]: When using `lang="en"` it contains quotes so need to switch to single quotes for that:

```suggestion
  caption='Fracción de solicitudes HTTP/2 o gQUIC de terceros conocidos por <i lang="en">content-type</i> por sitio web.',
  description='Un gráfico de barras que compara la fracción de solicitudes HTTP/2 de terceros por <i lang="en">content-type</i>. Todas las solicitudes de terceros se envían a través de HTTP/2 o gQUIC para el 90% de audio y fuentes, 80% de css y video, 70% de html, imágenes y texto y 60% de scripts.',
```

File: src/content/es/2020/http2.md [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+  sql_file="http2_1st_party_vs_3rd_party_by_type.sql"
+)
+}}
+
+Los <i lang="en">ads</i>, los servicios de <i lang="en">analytics</i>, los recursos de las redes de distribución de contenido (CDN), y los administradores de etiquetas se sirven principalmente a través de HTTP/2 o gQUIC, como se muestra en la Figura 22.9. Es más probable que el contenido de <i lang="en">customer-success</i> y de márketing se sirva a través de HTTP/1.
+
+{{ figure_markup(
+  image="http2-third-party-http2-usage-by-category.png",
+  caption="Fracción de solicitudes HTTP/2 o gQUIC de terceros conocidos por categoría por sitio web.",
+  description="Un gráfico de barras que compara la fracción de solicitudes HTTP/2 o gQUIC de terceros por categoría. Todas las solicitudes de terceros para todos los sitios web se atienden a través de HTTP/2 o gQUIC para el 95% de administradores de etiquetas, el 90% de servicios de <i lang="en">analytics</i> y de CDN, el 80% de <i lang="en">ads</i>, redes sociales, alojamiento y utilidades, el 40% de márketing y el 30% de <i lang="en">customer-success</i>.",
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:215]:
```suggestion
  description='Un gráfico de barras que compara la fracción de solicitudes HTTP/2 o gQUIC de terceros por categoría. Todas las solicitudes de terceros para todos los sitios web se atienden a través de HTTP/2 o gQUIC para el 95% de administradores de etiquetas, el 90% de servicios de <i lang="en">analytics</i> y de CDN, el 80% de <i lang="en">ads</i>, redes sociales, alojamiento y utilidades, el 40% de márketing y el 30% de <i lang="en">customer-success</i>.',
```

File: src/content/es/2020/http2.md [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+      <tr>
+        <td>Móvil</td>
+        <td class="numeric">62,849</td>
+        <td class="numeric">1.06%</td>
+        <td class="numeric">326</td>
+        <td class="numeric">0.06%</td>
+      </tr>
+    </tbody>
+  </table>
+  <figcaption>{{ figure_link(caption="Páginas que utilizan <i lang="en">server push</i> HTTP/2 o gQUIC.", sheets_gid="698874709", sql_file="number_of_h2_and_h3_pushed_resources_and_bytes_transferred.sql") }}</figcaption>
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:462]:
```suggestion
  <figcaption>{{ figure_link(caption='Páginas que utilizan <i lang="en">server push</i> HTTP/2 o gQUIC.', sheets_gid="698874709", sql_file="number_of_h2_and_h3_pushed_resources_and_bytes_transferred.sql") }}</figcaption>
```

File: src/content/es/2020/http2.md [hunk 4]
```diff
@@ -0,0 +1,690 @@
+---
+part_number: IV
+chapter_number: 22
+title: HTTP/2
+description: Capítulo sobre HTTP/2 del Web Alamanc de 2020 que cubre la adopción y el impacto  de HTTP/2, <i lang="en">HTTP/2 Push</i>, problemas con HTTP/2, y HTTP/3.
+authors: [dotjs, rmarx, MikeBishop]
+reviewers: [LPardue, bazzadp, ibnesayeed]
+analysts: [gregorywolf]
+translators: [moniloria]
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:9]: We added Editors recently:

```suggestion
editors: [rviscomi]
translators: [moniloria]
```

File: src/content/es/2020/http2.md [hunk 5]
```diff
@@ -0,0 +1,690 @@
...
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSOkWXtrbMfTLdhlKbBGDjRU3zKbnCQi3iPhfuKaFs5mj4smEzInDCYEnk63gBdgsJ3GFk2gf4FOKCU/pubchart?oid=1409316276&format=interactive",
+  sql_file="http2_1st_party_vs_3rd_party.sql",
+  sheets_gid="733872185"
+)
+}}
+¿Hay alguna diferencia ente cuáles tipos de contenido se sirven a través de HTTP/2 o gQUIC? La Figura 22.8 muestra, por ejemplo, que el 90% de los sitios web sirven el 100% de fuentes y audio de terceros a través de HTTP/2 o gQUIC, solo el 5% a través de HTTP/1.1 y un 5% son una combinación. La mayoría de los activos de terceros son scripts o imágenes, y se sirven únicamente a través de HTTP/2 o gQUIC en el 60% y el 70% de los sitios web, respectivamente.
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:198]: We like to keep line numbering consistent with English translation to make it easier to find the same lines for non-Spanish speakers (though looks like in a few places we'll need to fix the English version too - if you could add to this commit that would be great!)

```suggestion

¿Hay alguna diferencia ente cuáles tipos de contenido se sirven a través de HTTP/2 o gQUIC? La Figura 22.8 muestra, por ejemplo, que el 90% de los sitios web sirven el 100% de fuentes y audio de terceros a través de HTTP/2 o gQUIC, solo el 5% a través de HTTP/1.1 y un 5% son una combinación. La mayoría de los activos de terceros son scripts o imágenes, y se sirven únicamente a través de HTTP/2 o gQUIC en el 60% y el 70% de los sitios web, respectivamente.
```

File: src/content/es/2020/http2.md [hunk 6]
```diff
@@ -0,0 +1,690 @@
...
+
+Otro problema es que el servidor no conoce los recursos que el navegador ya ha almacenado en caché. Cuando un servidor intenta enviar algo que no es deseado, el cliente puede enviar una trama `RST_STREAM`, pero para cuando esto suceda, es posible que el servidor ya haya enviado todos los datos. Esto desperdicia ancho de banda y el servidor ha perdido la oportunidad de enviar inmediatamente algo que el navegador realmente requería. Hubo <a href="https://tools.ietf.org/html/draft-ietf-httpbis-cache-digest-05" hreflang="en">propuestas</a> para permitir a los clientes informar al servidor sobre el estado de su caché, pero tenían problemas de privacidad.
+
+Como se puede ver en la Figura 20.17 a continuación, un porcentaje muy pequeño de sitios utilizan <i lang="en">server push</i>.
+
+
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:433]: Remove one of the double blank lines to be consistent with English:

```suggestion
Como se puede ver en la Figura 20.17 a continuación, un porcentaje muy pequeño de sitios utilizan <i lang="en">server push</i>.

```

File: src/content/es/2020/http2.md [hunk 7]
```diff
@@ -0,0 +1,690 @@
...
+Finalmente, TLS ya se utiliza para proteger HTTP/1.1 y HTTP/2. Sin embargo, QUIC tiene una integración profunda de TLS 1.3, que protege tanto los datos HTTP/3 y los metadatos de los paquetes QUIC, como por ejemplo los números de paquete. El uso de TLS de esta manera mejora la privacidad y la seguridad del usuario final y facilita la evolución continua del protocolo. Combinar el transporte y los apretones de manos criptográficos significa que la configuración de la conexión requiere solo un RTT, en comparación con el mínimo de TCP de dos y en el peor de los casos de cuatro. En algunos casos, QUIC puede incluso ir un paso más allá y enviar datos HTTP junto con su primer mensaje, que se llama _0-RTT_. Se espera que estos tiempos rápidos de preparación de conexión realmente ayuden a HTTP/3 a superar a HTTP/2.
+
+
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:652]:
```suggestion

```

File: src/content/es/2020/http2.md [hunk 8]
```diff
@@ -0,0 +1,690 @@
...
+**Entonces, ¿HTTP/3 ayudará?**
+
+En la superficie, HTTP/3 no es realmente tan diferente de HTTP/2. No agrega ninguna característica importante, pero principalmente cambia la forma en que funcionan las características debajo de la superficie. Las mejoras reales provienen de QUIC, que ofrece configuraciones de conexión más rápidas, mayor solidez y resistencia a la pérdida de paquetes. Como tal, se espera que HTTP/3 funcione mejor que HTTP/2 en redes no tan buenas, mientras que ofrece un rendimiento muy similar en sistemas más rápidos. Sin embargo, eso es si la comunidad web puede hacer que HTTP/3 funcione, lo que puede ser un desafío en la práctica.
+
+
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:657]:
```suggestion

```

File: src/content/es/2020/http2.md [hunk 9]
```diff
@@ -0,0 +1,690 @@
...
+        <td class="numeric">453.57</td>
+      </tr>
+    </tbody>
+  </table>
+  <figcaption>{{ figure_link(caption="Distribución de activos empujados en dispositivos móviles.", sheets_gid="698874709", sql_file="number_of_h2_and_h3_pushed_resources_and_bytes_transferred.sql") }}</figcaption>
+</figure>
+
+Al observar la frecuencia de empuje por tipo de contenido en la Figura 22.20, vemos que el 90% de las páginas empujan scripts y el 56% empujan CSS. Esto tiene sentido, ya que pueden ser archivos pequeños que normalmente se encuentran en la ruta crítica para representar una página.
+
+{# TODO(analysts): Should this chart differentiate desktop and mobile? #}
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:573]: Looks like this was done. Can you remove this line from Spanish and English version?

```suggestion
```

File: src/content/es/2020/http2.md [hunk 10]
```diff
@@ -0,0 +1,690 @@
...
+  sql_file="adoption_of_http_2_by_site_and_requests.sql"
+)
+}}
+
+Es importante tener en cuenta que esta baja adopción solo refleja el uso de gQUIC y HTTP/3 para cargar páginas web. Desde hace varios años, Facebook ha tenido una implementación mucho más extensa de IETF QUIC y HTTP/3 para cargar datos dentro de sus aplicaciones nativas. <a href="https://engineering.fb.com/2020/10/21/networking-traffic/how-facebook-is-bringing-quic-to-billions/" hreflang="en">QUIC y HTTP/3 ya representan más del 75% de su tráfico total de Internet</a>. ¡Está claro que HTTP/3 recién está comenzando!
+
+Ahora podría preguntarse: si no todo el mundo está usando HTTP/2, ¿por qué necesitaríamos HTTP/3 tan pronto? ¿Qué beneficios podemos esperar en la práctica? Echemos un vistazo más de cerca a sus mecanismos internos.
+
+{# TODO(authors): I honestly found this section really interesting, but it doesn't contain any real-world data. How can we add an element of data to this? #}
+
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:637]: Looks like this was done. Could you remove from Spanish and English versions?

```suggestion
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Really good translation (at least technically - can't talk to the Spanish unfortunately!)

Spotted a few things minor things to fix though (which should also fix the reasons for the tests failing).

**Comment by moniloria** [comment]
> Thanks - will check it out!
> 
> > I also found some errors in the source content in English, who should I tell?
> 
> If they are small and you're reasonably certain they are errors (e.g. typos) can you add to this pull request? If you've questions or want to clarify something is correct [can you raise an issue](https://github.com/HTTPArchive/almanac.httparchive.org/issues) to discuss first?



> Thanks - will check it out!
> 
> > I also found some errors in the source content in English, who should I tell?
> 
> If they are small and you're reasonably certain they are errors (e.g. typos) can you add to this pull request? If you've questions or want to clarify something is correct [can you raise an issue](https://github.com/HTTPArchive/almanac.httparchive.org/issues) to discuss first?

Thanks for the review! I will modify the spaces and erase the "to do" from the english version as well.
The errors I found are only 1 typo (sever where it should say server), 2 errors in the figure description (copying the wrong numbers) and I also found a table that has all the numbers off! (figure 22.17 and 22.18 are the same!) I think I can fix the first 3 errors but I will raise an issue for the last one!

**Comment by bazzadp** [comment]
> I also found a table that has all the numbers off! (figure 22.17 and 22.18 are the same!)

Oh dunno how we missed that! I can see the new figures in the link sheets so let’s fix that here since you’re editing both files anyway. Gimme a few mins...

**Comment by moniloria** [comment]
> > I also found a table that has all the numbers off! (figure 22.17 and 22.18 are the same!)
> 
> Oh dunno how we missed that! I can see the new figures in the link sheets so let’s fix that here since you’re editing both files anyway. Gimme a few mins...

Oh ok! I have changed the other errors I've found.. can you confirm you see them? I will update when you're done fixing the tables!

File: src/content/es/2020/http2.md [hunk 11]
```diff
@@ -0,0 +1,686 @@
...
+        <td class="numeric">204.65</td>
+      </tr>
+      <tr>
+        <td class="numeric">90</td>
+        <td class="numeric">12</td>
+        <td class="numeric">351.05</td>
+        <td class="numeric">18</td>
+        <td class="numeric">453.57</td>
+      </tr>
+    </tbody>
```
Comment (bazzadp) [inline src/content/es/2020/http2.md:515]: Corrected desktop figures. Could you copy this change to English version too?

```suggestion
    <tbody>
      <tr>
        <td class="numeric">10</td>
        <td class="numeric">1</td>
        <td class="numeric">3.95</td>
        <td class="numeric">1</td>
        <td class="numeric">15.83</td>
      </tr>
      <tr>
        <td class="numeric">25</td>
        <td class="numeric">2</td>
        <td class="numeric">36.32</td>
        <td class="numeric">3</td>
        <td class="numeric">35.93</td>
      </tr>
      <tr>
        <td class="numeric">50</td>
        <td class="numeric">4</td>
        <td class="numeric">139.58</td>
        <td class="numeric">7</td>
        <td class="numeric">111.96</td>
      </tr>
      <tr>
        <td class="numeric">75</td>
        <td class="numeric">8</td>
        <td class="numeric">346.70</td>
        <td class="numeric">21</td>
        <td class="numeric">203.59</td>
      </tr>
      <tr>
        <td class="numeric">90</td>
        <td class="numeric">17</td>
        <td class="numeric">440.08</td>
        <td class="numeric">41</td>
        <td class="numeric">390.91</td>
      </tr>
    </tbody>
```

**Comment by moniloria** [comment]
I think it's all ready.. ! @bazzadp

**Review by bazzadp — APPROVED** [review]
LGTM! Thanks for working on this - and for fixing the English version!

Ideally one of the other @HTTPArchive/spanish-translators will give the language a once over but if I don't hear anything by time of next release then will merge this in. Have a few outstanding pull requests there so will wait and see if we can merge any of those over next few days before doing a release.

**Pull request merged**